src/Drupal/Commands/core/LocaleCommands.php

```diff
@@ -128,4 +133,183 @@ public function update($options = ['langcodes' => self::REQ])
 
         drush_backend_batch_process();
     }
+
+    /**
+     * Exports to a gettext translation file.
+     *
+     * @see \Drupal\locale\Form\ExportForm::submitForm
+     *
+     * @throws \Exception
+     *
+     * @command locale:export
+     * @drupal-dependencies locale
+     * @param $langcode The language code of the exported translations.
+     * @option template POT file output of extracted source texts to be translated.
+     * @option types String types to include, defaults to all types. Types: 'not-customized', 'customized', 'not-translated'.
+     * @usage drush locale:export nl > nl.po
+     *   Export the Dutch translations with all types.
+     * @usage drush locale:export nl --types=customized,not-customized > nl.po
+     *   Export the Dutch customized and not customized translations.
+     * @usage drush locale:export --template > drupal.pot
+     *   Export the basic template file to translate.
+     * @usage drush locale:export --template nl > nl.pot
+     *   Export the Dutch template file to translate.
+     * @aliases locale-export
+     */
+    public function export($langcode = null, $options = ['template' => false, 'types' => self::OPT])
+    {
+        $language = $this->getTranslatableLanguage($langcode);
+        $poreader_options = [];
+
+        if (!(bool)$options['template']) {
+            $poreader_options = $this->convertTypesToPoDbReaderOptions(StringUtils::csvToArray($options['types']));
+        }
+
+        $file_uri = drush_save_data_to_temp_file('temporary://', 'po_');
+        if ($this->writePoFile($file_uri, $language, $poreader_options)) {
+            $this->printFile($file_uri);
+        } else {
+            $this->logger()->notice(dt('Nothing to export.'));
+        }
+    }
+
+    /**
+     * Assure that required options are set.
+     *
+     * @hook validate locale:export
+     */
+    public function exportValidate(CommandData $commandData)
+    {
+        $langcode = $commandData->input()->getArgument('langcode');
+        $template = $commandData->input()->getOption('template');
+        $types = $commandData->input()->getOption('types');
+
+        if (!$langcode && !$template) {
+            throw new \Exception(dt('Set LANGCODE or --template, see help for more information.'));
+        }
+        if ($template && $types) {
+            throw new \Exception(dt('No need for --types, when --template is used, see help for more information.'));
+        }
+    }
+
+    /**
+     * Get translatable language object.
+     *
+     * @param string $langcode The language code of the language object.
+     * @return LanguageInterface|null
+     * @throws \Exception
+     */
+    private function getTranslatableLanguage($langcode)
+    {
+        if (!$langcode) {
+            return null;
+        }
+
+        $language = \Drupal::languageManager()->getLanguage($langcode);
```

Comment:
You should be able to use dependency injection to get the languageManager(). Also Drupal::config() later.